### PR TITLE
fix(Core/Spells): Ensure gameobjects are still locked/closed when loc…

### PR DESCRIPTION
**Pull request by Nyeriah**
…kpicking/opening them

## Changes Proposed:
-  Check if gameobjects still have state GO_STATE_ACTIVE set (not open or destroyed) by the time spells with effect SPELL_OPEN_LOCK finish.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7446
- closes https://github.com/chromiecraft/chromiecraft/issues/1461
- 
## Tests Performed:

- Builds successfully on Win11.
- Tested several spells affected by this, none of them presented any issues (e.g Eye of Eternity Focusing Iris Key, tested with several encounter resets even)
- Tested opening locked chests, gathering profession mats (herbs, mining), opening lootboxes (items) - no issues presented.
- Tested Deadmines Iron Clad Door.

## How to Test the Changes:

1. Create two characters (preferrably rogues)
2. .go obj 30534
3. .learn 9133 (lockpicking) & .max all
4. Tab to the other character, give yourself item 5397 (Defias Gunpowder)
5. Use the cannon & lockpick the door simultaneously, so by the time the lockpicking spell finishes the door is already destroyed.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/Spell.cpp [hunk 1]
```diff
@@ -5815,6 +5815,11 @@ SpellCastResult Spell::CheckCast(bool strict)
                             (!pTempItem || !pTempItem->GetTemplate()->LockID || !pTempItem->IsLocked()))
                         return SPELL_FAILED_BAD_TARGETS;
 
+                    // We must also ensure the gameobject we are opening is still closed by the time the spell finishes.
+                    if (GameObject* go = m_targets.GetGOTarget())
+                        if (go->GetGoState() != GO_STATE_READY)
+                            return SPELL_FAILED_BAD_TARGETS;
+
```
Comment (Helias) [inline src/server/game/Spells/Spell.cpp:5822]: add brackets
```suggestion
                    if (GameObject* go = m_targets.GetGOTarget())
                    {
                        if (go->GetGoState() != GO_STATE_READY)
                        {
                            return SPELL_FAILED_BAD_TARGETS;
                        }
                    }
```

**Comment by jamadaha** [comment]
Tested:
The door now stays open even after the lock picker finishes.
So seems to work as expected.

File: src/server/game/Spells/Spell.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/game/Spells/Spell.cpp:5822]: Is that the oficial AC code aesthetic standard? I'm just asking, I'm used to TC standard and we don't use so many brackets. I can understand bracketing a dubious IF, but it's not the case here and it really leaves me wondering.

Comment (Helias) [reply src/server/game/Spells/Spell.cpp:5822]: yes it is https://www.azerothcore.org/wiki/cpp-code-standards

**Review by Helias — APPROVED** [review]

**Pull request merged**